packages/core/contracts/financial-templates/optimistic-rewarder/OptimisticRewarder.sol

```diff
@@ -0,0 +1,429 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+import "../../common/implementation/Lockable.sol";
+import "../../common/implementation/Testable.sol";
+import "../../oracle/interfaces/FinderInterface.sol";
+import "../../oracle/interfaces/StoreInterface.sol";
+import "../../oracle/interfaces/IdentifierWhitelistInterface.sol";
+import "../../oracle/implementation/Constants.sol";
+import "../../oracle/interfaces/SkinnyOptimisticOracleInterface.sol";
+import "../../common/implementation/AncillaryData.sol";
+import "../../common/interfaces/AddressWhitelistInterface.sol";
+import "./OptimisticRewarderToken.sol";
+
+/**
+ * @notice The base rewarder contract. This manages depositing rewards and paying them out to tokenholders using values
+ * backed by the OptimisticOracle's dispute process.
+ */
+abstract contract OptimisticRewarderBase is Lockable {
```

Comment:
should this be placed in a separate file?

Reply:
Sure, was trying to keep the changes _looking_ small. I can do that :)